src/lib.rs

```diff
@@ -136,6 +144,98 @@ where
     }
 }
 
+impl<P, T> Atom<P>
+where
+    P: IntoRawPtr + FromRawPtr + Deref<Target = T>,
+{
+    /// Stores a value into the pointer if the current value is the same as the
+    /// current value.
+    ///
+    /// The return value is always the previous value. If it is equal to
```

Comment:
This sentence doesn't make any sense to me, you can only return the previous value if you swapped it out.

Reply:
This needs to include that

1) on success, the previous value is returned
2) on failure, the `new` value is returned together with a raw pointer of `current` which you must not read (please make this very clear in the docs)

Reply:
read=dereference